saleor/plugins/subbly/plugin.py

```
@@ -83,12 +84,15 @@ def webhook(self, request: WSGIRequest, path: str, previous_value) -> HttpRespon
                 customer = payload.get("customer").get("user", {})
 
                 # Create subscription object
-                subscription = SubblySubscription.objects.create(
-                    subscription_id=payload.get("id"),
-                    first_name=customer.get("first_name"),
-                    last_name=customer.get("last_name"),
-                    email=customer.get("email"),
-                )
+                try:
+                    subscription = SubblySubscription.objects.create(
+                        subscription_id=payload.get("id"),
+                        first_name=customer.get("first_name"),
+                        last_name=customer.get("last_name"),
+                        email=customer.get("email"),
+                    )
+                except IntegrityError:  # fail silently
+                    return JsonResponse({"status": "email already registered"})
```

@garguelles does this error get logged anywhere? are there details in the IntegrityError message that might be useful? This feels like a "code smell" to me. If something goes wrong with the DB, like foreign key check fails, duplicate key, etc. will we ever know about it?

---

hi @jsnbuchanan, This is the subbly webhook listener.  This try catch block is scoped for the creation of subbly subscriptions, the only integrity error that can happen here is if an email already exist in our subbly subscriptions table. I just decided to have this fail silently as this is not really an error, more of a validation flow. 

I could do a query to check if it exists but that means this webhook will do 2 queries instead of one 😄